pkg/report/table/table.go

```diff
@@ -45,55 +45,85 @@ type Writer struct {
 	// For licenses
 	LicenseRiskThreshold int
 	IgnoredLicenses      []string
+
+	// We need to use the same renderers for all results, for the correct operation of logs, once functions, etc.
+	renderers
 }
 
 type Renderer interface {
-	Render() string
+	Render(result types.Result)
+	Flush() string
+}
+
+type renderers struct {
+	vulnerabilityRenderer *vulnerabilityRenderer
+	misconfigRenderer     *misconfigRenderer
+	secretRenderer        *secretRenderer
+	pkgLicenseRenderer    *pkgLicenseRenderer
+	fileLicenseRenderer   *fileLicenseRenderer
+}
+
+func (tw *Writer) initRenderers() {
+	tw.renderers = renderers{
+		vulnerabilityRenderer: NewVulnerabilityRenderer(tw.isOutputToTerminal(), tw.Tree, tw.ShowSuppressed, tw.Severities),
+		misconfigRenderer:     NewMisconfigRenderer(tw.Severities, tw.Trace, tw.IncludeNonFailures, tw.isOutputToTerminal()),
+		secretRenderer:        NewSecretRenderer(tw.isOutputToTerminal(), tw.Severities),
+		pkgLicenseRenderer:    NewPkgLicenseRenderer(tw.isOutputToTerminal(), tw.Severities),
+		fileLicenseRenderer:   NewFileLicenseRenderer(tw.isOutputToTerminal(), tw.Severities),
+	}
 }
 
 // Write writes the result on standard output
-func (tw Writer) Write(_ context.Context, report types.Report) error {
+func (tw *Writer) Write(_ context.Context, report types.Report) error {
+	tw.initRenderers()
 
 	for _, result := range report.Results {
 		// Not display a table of custom resources
 		if result.Class == types.ClassCustom {
 			continue
 		}
-		tw.write(result)
+		tw.render(result)
 	}
+
+	tw.flush()
 	return nil
 }
 
-func (tw Writer) write(result types.Result) {
+func (tw *Writer) flush() {
+	_, _ = fmt.Fprint(tw.Output, tw.vulnerabilityRenderer.Flush())
+	_, _ = fmt.Fprint(tw.Output, tw.misconfigRenderer.Flush())
+	_, _ = fmt.Fprint(tw.Output, tw.secretRenderer.Flush())
+	_, _ = fmt.Fprint(tw.Output, tw.pkgLicenseRenderer.Flush())
+	_, _ = fmt.Fprint(tw.Output, tw.fileLicenseRenderer.Flush())
```

Comment:
I think we can pass bytes.Buffer to each renderer rather than each renderer holding its own buffer. The buffer will be flushed at the end.

Reply:
Yes 👍